pyof/foundation/network_types.py

```diff
@@ -9,7 +9,94 @@
 from pyof.foundation.constants import VLAN_TPID
 from pyof.foundation.exceptions import PackException, UnpackException
 
-__all__ = ('Ethernet', 'GenericTLV', 'IPv4', 'VLAN', 'TLVWithSubType', 'LLDP')
+__all__ = ('ARP', 'Ethernet', 'GenericTLV', 'IPv4', 'VLAN', 'TLVWithSubType',
+           'LLDP')
+
+
+class ARP(GenericStruct):
+    """ARP packet "struct".
+
+    Contains fields for an ARP packet's header and data.
+    Designed for Ethernet and IPv4 only: needs to have some attributes changed
+    for other HTYPE and PTYPE implementations.
+    Must be encapsulated inside an Ethernet frame.
+    """
+
+    #: htype (:class:`UBInt16`): The L2 (hardware) protocol type.
```

Comment:
Comments in both class and instance attributes generate too much output in doc (and are also boring for us to duplicate stuff :)). A long time ago (I had to check some old commits), we decided to write comments about attributes like these only in ``__init__``'s ``Args:`` list. Here, it seems to be enough only to remove the class attributes' comments. I'm citing @diraol to refresh his memory and also @macartur, the Sphinx Man because I don't recall telling him about this. No need to fix merged code, but keep this is mind.

Note: It is redundant (both in code and HTML result) to write attr name and also type in this case. Sphinx infers them from the line immediately below. Thus, ``#: The L2 (hardware) protocol type.`` would suffice.